pkg/volume/util.go

```diff
@@ -142,8 +142,9 @@ func (c *realRecyclerClient) WatchPod(name, namespace string, stopChannel chan s
 	cache.NewReflector(podLW, &api.Pod{}, queue, 1*time.Minute).RunUntil(stopChannel)
 
 	return func() *api.Pod {
-		obj := queue.Pop()
-		return obj.(*api.Pod)
+		var pod *api.Pod
+		queue.Pop(func(obj interface{}) error { pod = obj.(*api.Pod); return nil })
```

Comment:
Pop returns an object - should we simply have Pop stop returning an object, or should we have two methods `Pop() interface{}` and `PopAndLock(func ...)`, or should we add a helper function which is a no-op `cache.Ok = func() {}`?